examples/solids/src/cloptions.c

```diff
@@ -91,6 +109,37 @@ PetscErrorCode ProcessCommandLineOptions(MPI_Comm comm, AppCtx appCtx) {
                             NULL, appCtx->bcClampMax, &appCtx->bcClampMax,
                             NULL); CHKERRQ(ierr);
 
+  appCtx->bcTwistXCount = 16;
+  ierr = PetscOptionsIntArray("-bc_twist_x",
+                              "Face IDs to apply twisted Dirichlet BC",
+                              NULL, appCtx->bcTwistXFaces,
+                              &appCtx->bcTwistXCount, NULL); CHKERRQ(ierr);
+  appCtx->bcTwistXMax = 1;
```

Comment:
So long as faces are numbered, you could process `-bc_twist_998_rotate .3,.6,.9` where the arguments are a (Rodrigues vector](https://en.wikipedia.org/wiki/Rodrigues%27_rotation_formula. This is a relatively compact and intuitive method. You could scale the vector by pi to make it easier to specify even fractions.

Reply:
The translation option was very quick to code up.

For rotation, do we want the users to provide `R = I + (sin theta) K + (1 - cos theta) K^2` (what I understand the Rodriques vector to be) or would it be more intuitive for them to specify an axis and angle?

Reply:
Ah, it might not be clear in the article, but the usual approach is that you specify a single (non-unit) vector `k`. The orientation of `k` defines the axis and its length is `\theta` (how much to rotate around that axis). We could choose to scale by pi (or even pi/180) to make normalization more convenient for axis-aligned rotation. When off-axis, you generally need to do some math separately, but I suppose the command line could accept `kx,ky,kz,theta` which would mean to normalize the specified vector `k = (kx, ky, kz)` and scale it by `theta`. If using degrees, that would allow `-bc_clamp_998_rotate 3,4,5,45` (rotate by 45 degrees about the specified axis).

Reply:
I think `kx,ky,kz,theta` would make it easier for users to tinker or adjust bc parameter values, at least for me.

Reply:
Sounds good to me. You could allow `kx,ky,kz` (theta = norm(k)) as short-hand, or skip it for uniformity.